view/js/linkPreview.js

```diff
@@ -1,877 +1,881 @@
-			firstPosted = false;
-			isActive = false;
-			isCrawling = false;
-			selector = "";
-		};
-
-		var trim = function(str) {
-			return str.replace(/^\s+|\s+$/g, "");
-		};
-		var escapeHTML = function(unsafe_str) {
-			return unsafe_str
-				.replace(/&/g, '&amp;')
-				.replace(/</g, '&lt;')
-				.replace(/>/g, '&gt;')
-				.replace(/\"/g, '&quot;')
-				.replace(/\[/g, '&#91;')
-				.replace(/\]/g, '&#93;')
-				.replace(/\'/g, '&#39;'); // '&apos;' is not valid HTML 4
-		};
-
-		// Initialize LinkPreview 
-		init();
-
-		return {
-			// make crawlText() accessable from the outside.
-			crawlText: function(text) {
-				crawlText(text);
-			},
-			addBBCodeToPreview: function(content) {
-				addBBCodeToPreview(content);
-			},
-			destroy: function() {
-				destroy();
-			}
-		};
-	};
-
-	$.fn.linkPreview.defaults = {
-		defaultDescription: "Enter a description",
-		defaultTitle: "Enter a title"
-	};
-
-	/**
-	* Get in a textarea the previous word before the cursor.
-	* 
-	* @param {object} text Textarea elemet.
-	* @param {integer} caretPos Cursor position.
-	* 
-	* @returns {string} Previous word.
-	*/
-	function returnWord(text, caretPos) {
-		var index = text.indexOf(caretPos);
-		var preText = text.substring(0, caretPos);
-		// If the last charachter is a space or enter remove it
-		// We need this in friendica for the url  preview.
-		var lastChar = preText.slice(-1)
-		if ( lastChar === " "
-			|| lastChar === "\n"
-			|| lastChar === "\r"
-			)
-		{
-			preText = preText.substring(0, preText.length -1);
-		}
-
-		// Replace new line with space.
-		preText = preText.replace(/\n/g, " ");
-
-		if (preText.indexOf(" ") > 0) {
-			var words = preText.split(" ");
-			return words[words.length - 1]; //return last word
-		}
-		else {
-			return preText;
-		}
-	}
-
-	/**
-	 * Get in a textarea the previous word before the cursor.
-	 * 
-	 * @param {string} id The ID of a textarea element.
-	 * @returns {sting|null} Previous word or null if no word is available.
-	 */
-	function getPrevWord(id) {
-		var text = document.getElementById(id);
-		var caretPos = getCaretPosition(text);
-		var word = returnWord(text.value, caretPos);
-		if (word != null) {
-			return word
-		}
-
-	}
-
-	/**
-	 * Get the cursor posiotion in an text element.
-	 * 
-	 * @param {object} ctrl Textarea elemet.
-	 * @returns {integer} Position of the cursor.
-	 */
-	function getCaretPosition(ctrl) {
-		var CaretPos = 0;   // IE Support
-		if (document.selection) {
-			ctrl.focus();
-			var Sel = document.selection.createRange();
-			Sel.moveStart('character', -ctrl.value.length);
-			CaretPos = Sel.text.length;
-		}
-		// Firefox support
-		else if (ctrl.selectionStart || ctrl.selectionStart == '0') {
-			CaretPos = ctrl.selectionStart;
-		}
-		return (CaretPos);
-	}
-})(jQuery);
+// @license magnet:?xt=urn:btih:d3d9a9a6595521f9666a5e94cc830dab83b65699&dn=expat.txt Expat
+// @license magnet:?xt=urn:btih:0b31508aeb0634b347b8270c7bee4d411b5d4109&dn=agpl-3.0.txt GOL
```

Comment:
What's the `GOL` license? Isn't it supposed to be `AGPLv3-or-later` as well?

Reply:
Thanks for spotting that - copy and paste issue I think. Should be GPL